src/main/scala/scalismo/mesh/boundingSpheres/SurfaceSpatialIndex.scala

```diff
@@ -39,24 +43,33 @@ trait SurfaceSpatialIndex[D] {
    * @return The closest point and the squared distance.
    */
   def getClosestPoint(pt: Point[D]): ClosestPoint
+}
+
+/**
+ * SurfaceDistance trait with the basic queries defined.
+ */
+trait SurfaceSpatialIndex[D] extends SpatialIndex[D] {
 
   /**
    * Query the closest point on a surface.
    *
    * @return A desciption of the closest point.
    */
-  def getClosestPointOnSurface(pt: Point[D]): ClosestPointOnSurface
+  def getClosestPointOnSurface(pt: Point[D]): ClosestPointWithSquaredDistance
 }
 
-/**
- * Type of the closest point. At the moment the names are only suited for a triangular mesh.
- */
-private object ClosestPointType extends Enumeration {
-  type ClosestPointType = Value
-  val POINT, ON_LINE, IN_TRIANGLE = Value
+object SurfaceSpatialIndex {
+
+  /**
+   * Type of the closest point. At the moment the names are only suited for a triangular mesh.
+   */
+  private[boundingSpheres] object SurfaceClosestPointType extends Enumeration {
+    type SurfaceClosestPointType = Value
```

Comment:
Maybe ClosestPointType would be sufficient? It is clear from the surrounding context that it refers to the ClosestPointType of the surface.

Reply:
There are two versions of those. `SurfaceClosestPointType` and `VolumeClosestPointType` exists both. But I will remove the surrounding object which is not needed.

Reply:
Object around removed but kept the name.